dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamBuiltinAggregations.java

```diff
@@ -383,6 +384,144 @@ public T result(KV<BigDecimal, Long> accumulator) {
     }
   }
 
+  static class VarAgg implements Serializable {
+    long count; // number of elements
+    double sum; // sum of elements
+
+    public VarAgg(long count, double sum) {
+      this.count = count;
+      this.sum = sum;
+    }
+  }
+
+  public static final class Var<T> extends BeamSqlUdaf<T, KV<BigDecimal, VarAgg>, T> {
```

Comment:
We should follow the constrain of Calcite, otherwise it cannot even pass compile/validation.